File: KisaragiTech.Dape/User/Service/UserIDGenerationService.cs
using System;
using KisaragiTech.Dape.User.Model;

namespace KisaragiTech.Dape.User.Service;

public static class UserIDGenerationService
{
    public static UserID Generate()
    {
        return new UserID(Guid.NewGuid());
    }
}

File: src/main/scala/io/github/ksrgtech/dape/user/service/UserIdGenerationService.scala
package io.github.ksrgtech.dape.user.service

import cats.effect.kernel.Sync
import io.github.ksrgtech.dape.user.model.UserId

import java.util.UUID

object UserIdGenerationService {

  def generate[F[_]: Sync]: F[UserId] =
    Sync[F].delay {
      UserId(UUID.randomUUID())
    }
}
